h_malloc.c

```diff
@@ -2152,7 +2157,20 @@ COLD EXPORT int h_malloc_set_state(UNUSED void *state) {
 #ifdef __ANDROID__
 COLD EXPORT void h_malloc_disable_memory_tagging(void) {
 #ifdef HAS_ARM_MTE
-    __is_memtag_enabled = false;
+    if (!ro.is_memtag_disabled) {
```

Comment:
Seems like a spot for `is_memtag_enabled`?

Reply:
I've inlined `is_memtag_enabled()` here to make clearer how ` h_malloc_disable_memory_tagging()` works.